driver-core/src/main/java/com/datastax/driver/core/JdkSSLOptions.java

```diff
@@ -0,0 +1,115 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.security.NoSuchAlgorithmException;
+
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.ssl.SslHandler;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+
+/**
+ * {@link SSLOptions} implementation based on built-in JDK classes.
+ */
+public class JdkSSLOptions implements SSLOptions {
+
+    /**
+     * Creates a builder to create a new instance.
+     *
+     * @return the builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private final SSLContext context;
+    private final String[] cipherSuites;
+
+    protected JdkSSLOptions(SSLContext context, String[] cipherSuites) {
+        this.context = (context == null) ? makeDefaultContext() : context;
+        this.cipherSuites = cipherSuites;
+    }
+
+    @Override
+    public SslHandler newSSLHandler(SocketChannel channel) {
+        SSLEngine engine = newSSLEngine();
+        return new SslHandler(engine);
+    }
+
+    protected SSLEngine newSSLEngine() {
```

Comment:
Since this method is protected, it will appear in the javadocs, so it would be better if there was a short description of situations where users could override it (e.g. hostname verification, see JAVA-716).